### DISCUSSION TITLE
Calculate GND from grains structure

### DISCUSSION THREAD

**Discussion by JinghaoXu92** [mtex-toolbox/mtex#1295] [Ask Anything] [answered]
Hello,

From ([https://mtex-toolbox.github.io/GND.html]):
`plot(ebsd,factor*sum(abs(rho .* dSRot.u),2))`
It presents the GND from grid to grid. And 'factor*sum(abs(rho .* dSRot.u),2)' sorted in one column.
The calculations seem based on the _ebsd structure_.
What I was wondering is if this can be done from _grains structure_?
For this moment, I am using e.g.:
`ebsd_grain1 = ebsd(grains( grains.id ==  1))`
and then start the procedures from [https://mtex-toolbox.github.io/GND.html] manually for 'grains.id ==  1'.

Do you see there're some more effective ways to do this?

It's will be much appreciated!

BTW, at that page there's a comment:
```
% in bcc and fcc the norm of the burgers vector is sqrt(3)/2 * a
[norm(dS(1).b), norm(dS(end).b), sqrt(3)/2 * a]
```

But for fcc, would this value be sqrt(2)/2 * a rather than sqrt(3)/2 * a?

Thank you in advance!

Cheers,
Jinghao

**Comment by kilir** [comment]
Hi,
maybe it may help if you'd explain what you would want to achieve, i.e. why computing the GND from a subset of your EBSD would be different or beneficial for you than computing it from the entire set.

Cheers,
Rüdiger

**Reply by JinghaoXu92** [reply]
Hi,

I am doing post-mortem observation on the deformed samples. The dislocations distribution is heterogeneous dependent on the crystal orientation. What I want to examine is the dislocation density as a function of the Schmid factor.
The Schmid factor is calculated from meanOrientation of individual grains. So I want to know GND from the corresponding grains as well.
![image](https://user-images.githubusercontent.com/29089463/144755000-36372788-d281-4a1d-8b8b-1ee1968391e7.png)
The above shows the expected plots. Based on this, some conclusions like 'dislocation mediated plasticity' can be roughly drawn.

Hope this would make the request clearer.

Cheers,
Jinghao

**Comment by zmichels** [comment, marked as answer]
I don’t think you *need* to use the grain mean orientation for Schmid factor (unless you specific want a grain-scale Schmid factor). You can compute it for every solution/pixel… and then do the same for your GND calcs… and also have a more-robust dataset to work with.

Zach

**Reply by JinghaoXu92** [reply]
Thank you for this input. It's a good point.
Also agree with the 'more-robust dataset'. Mean orientation averaged all pixels, for coarse and or deformed grain in this case, they may be varied a lot.

/Jinghao
